File: smartadapter/src/test/java/smartadapter/viewholders/ViewAttachedToWindowTestViewHolder.java
package smartadapter.viewholders;

/*
 * Created by Manne Öhlund on 2019-07-16.
 * Copyright (c) All rights reserved.
 */

import android.view.View;

import smartadapter.listener.OnViewAttachedToWindowListener;
import smartadapter.listener.OnViewDetachedFromWindowListener;
import smartadapter.viewholder.SmartViewHolder;

public class ViewAttachedToWindowTestViewHolder extends SmartViewHolder implements OnViewAttachedToWindowListener, OnViewDetachedFromWindowListener {

    public ViewAttachedToWindowTestViewHolder(View view) {
        super(view);
    }

    @Override
    public void bind(Object item) {

    }

    @Override
    public void onViewAttachedToWindow() {

    }

    @Override
    public void onViewDetachedFromWindow() {

    }
}

File: smartadapter/src/test/java/smartadapter/viewholders/ViewAttachedToWindowTestViewHolder.kt
package smartadapter.viewholders

/*
 * Created by Manne Öhlund on 2019-07-16.
 * Copyright (c) All rights reserved.
 */

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import smartadapter.listener.OnViewAttachedToWindowListener
import smartadapter.listener.OnViewDetachedFromWindowListener
import smartadapter.viewholder.SmartViewHolder

open class ViewAttachedToWindowTestViewHolder(view: View) : SmartViewHolder<Any>(view),
    OnViewAttachedToWindowListener,
    OnViewDetachedFromWindowListener {

    override fun bind(item: Any) { }

    override fun onViewAttachedToWindow(viewHolder: RecyclerView.ViewHolder) { }

    override fun onViewDetachedFromWindow(viewHolder: RecyclerView.ViewHolder) { }
}
